cmd/crank/render/runtime_docker.go

```diff
@@ -162,8 +163,24 @@ func GetRuntimeDocker(fn pkgv1.Function, log logging.Logger) (*RuntimeDocker, er
 		return nil, errors.Wrapf(err, "cannot get pull policy for Function %q", fn.GetName())
 	}
 
-	// Initial ConfigFile
-	configFile := config.LoadDefaultConfigFile(os.Stderr)
+	// Initial ConfigFile, first check environment variable XDG_RUNTIME_DIR for Podman if it exists
+	// Otherwise, use the default Docker config file
+	var configFile *configfile.ConfigFile
+	if _, err := os.Stat(filepath.Join(os.Getenv("XDG_RUNTIME_DIR"), "containers/auth.json")); err == nil {
```

Comment:
i think we could refactor this logic to its own function, then we can streamline the ordering and use the guard clause / return early pattern for readability and error handling. I pushed an example commit of this refactor to https://github.com/crossplane/crossplane/compare/main...jbw976:crossplane:render-podman for you to check out

Reply:
@jbw976 Thanks for the example! After look into the `go-containerregistry` library, I think it would be good to keep using the same library between `render` and `beta validate`. So we make sure they're using the same logic to handle credentials for pulling private images.